modules/openxr/extensions/openxr_composition_layer_extension.cpp

```diff
@@ -143,42 +203,117 @@ void OpenXRViewportCompositionLayerProvider::set_alpha_blend(bool p_alpha_blend)
 }
 
 void OpenXRViewportCompositionLayerProvider::set_viewport(RID p_viewport, Size2i p_size) {
+	ERR_FAIL_COND(use_android_surface);
+
 	RenderingServer *rs = RenderingServer::get_singleton();
 	ERR_FAIL_NULL(rs);
 
-	if (viewport != p_viewport) {
-		if (viewport.is_valid()) {
-			RID rt = rs->viewport_get_render_target(viewport);
+	if (subviewport.viewport != p_viewport) {
+		if (subviewport.viewport.is_valid()) {
+			RID rt = rs->viewport_get_render_target(subviewport.viewport);
 			RSG::texture_storage->render_target_set_override(rt, RID(), RID(), RID());
 		}
 
-		viewport = p_viewport;
+		subviewport.viewport = p_viewport;
 
-		if (viewport.is_valid()) {
-			viewport_size = p_size;
+		if (subviewport.viewport.is_valid()) {
+			subviewport.viewport_size = p_size;
 		} else {
 			free_swapchain();
-			viewport_size = Size2i();
+			subviewport.viewport_size = Size2i();
 		}
 	}
 }
 
+void OpenXRViewportCompositionLayerProvider::set_use_android_surface(bool p_use_android_surface, Size2i p_size) {
+#ifdef ANDROID_ENABLED
+	if (p_use_android_surface != use_android_surface) {
```

Comment:
I'd prefer for this to check for equality and then (if true) do an early return, rather than nest everything inside another if statement.

Reply:
Similar thing [here](https://github.com/godotengine/godot/blob/864851b9a9091d242af41bb72f7912303f8eaa10/modules/openxr/scene/openxr_composition_layer.cpp#L267) as well.